core/drand.go

```diff
@@ -92,19 +96,41 @@ func initDrand(s key.Store, c *Config) (*Drand, error) {
 	d.callbacks.AddCallback(callbackID, d.opts.callbacks)
 	//d.callbacks.AddCallback(cacheID, d.cache.StoreTemp)
 
-	a := c.ListenAddress(priv.Public.Address())
+	// Set the private API address to the command-line flag, if given.
+	// Otherwise, set it to the address associated with stored private key.
+	privAddr := c.PrivateListenAddress(priv.Public.Address())
+	pubAddr := c.PublicListenAddress("")
 	if c.insecure {
+		var err error
 		d.log.Info("network", "tls-disable")
-		d.gateway = net.NewGrpcGatewayInsecure(a, d, d.opts.grpcOpts...)
+		if pubAddr != "" {
+			if d.pubGateway, err = net.NewRESTPublicGatewayWithoutTLS(d.ctx, pubAddr, d, d.opts.grpcOpts...); err != nil {
```

Comment:
It shouldn't need the grpcOptions there since it's explicitily a rest api now, right ?

Reply:
(This is not the latest code.)